src/components/WalletDropdown/MiniPortfolio/Activity/parseLocal.ts

```diff
@@ -182,24 +181,20 @@ export function parseLocalActivity(
   return { ...defaultFields, ...additionalFields }
 }
 
-export function useLocalActivities(): ActivityMap | undefined {
+export function useLocalActivities(account: string): ActivityMap {
   const allTransactions = useMultichainTransactions()
-  const { chainId } = useWeb3React()
   const tokens = useCombinedActiveList()
 
   return useMemo(
     () =>
-      chainId
-        ? allTransactions.reduce((acc: { [hash: string]: Activity }, [transaction, chainId]) => {
-            try {
-              const localActivity = parseLocalActivity(transaction, chainId, tokens)
-              if (localActivity) acc[localActivity.hash] = localActivity
-            } catch (error) {
-              console.error('Failed to parse local activity', transaction)
-            }
-            return acc
-          }, {})
-        : undefined,
-    [allTransactions, chainId, tokens]
+      allTransactions.reduce((acc: { [hash: string]: Activity }, [transaction, chainId]) => {
```

Comment:
i also feel like this `.reduce` can just be turned into a for loop for readability. since you're just modifying the `acc` object in place this isn't a true reduce in the sense that it's a pure function 

would probably be better to do something like
```
const activityByHash = {}

for (const transaction of allTransactions) {
   // add to activityByHash
}

return activityByHash

Reply:
ty for this. I've been using reduce way too much for no good reason, this is way cleaner